File: webui/src/types/modal/character.ts
export type AddCharacterForm = {
  name: string | null
  description: string | null
  content: { key: string; value: string }[]
  settings: object | null
  isPublic: boolean
  avatar: number | null
}

export type EditCharacterForm = {
  id: number
  name: string
  description: string | null
  content: { key: string; value: string }[]
  settings: object | null
  isPublic: boolean
  avatar: number | null
}

File: src/Models/Character.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace LLMRolePlay.Models
{
  public partial class DBContext : Microsoft.EntityFrameworkCore.DbContext
  {
    public DbSet<Character> Characters { get; set; }
  }
  public class Character
  {
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public uint Id { get; set; }
    public required string Name { get; set; }
    public string? Settings { get; set; }
    public required string Content { get; set; }
    public string? Description { get; set; }
    public bool IsPublic { get; set; }
    public uint UserId { get; set; }

    public uint? Avatar { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public async Task Delete(DBContext db)
    {
      db.Characters.Remove(this);
      await db.SaveChangesAsync();
    }
    public void MarkAsModified(DBContext db)
    {
      db.Entry(this).State = EntityState.Modified;
    }
  }
}